onfido-java/src/main/java/com/onfido/api/ResourceManager.java

```diff
@@ -35,74 +26,63 @@ public static void shutdown() {
     CLIENT.connectionPool().evictAll();
   }
 
-  public String post(String path, String body) throws OnfidoException {
+  protected String post(String path, String body) throws OnfidoException {
     Request request = requestBuilder(path)
       .post(RequestBody.create(body, JSON))
       .build();
 
     return performRequest(request);
   }
 
-  public String get(String path) throws OnfidoException {
+  protected String get(String path) throws OnfidoException {
     Request request = requestBuilder(path).build();
 
     return performRequest(request);
   }
 
-  public String put(String path, String body) throws OnfidoException {
+  protected String put(String path, String body) throws OnfidoException {
     Request request = requestBuilder(path)
             .put(RequestBody.create(body, JSON))
             .build();
 
     return performRequest(request);
   }
 
-  public String deleteRequest(String path) throws OnfidoException {
+  protected String deleteRequest(String path) throws OnfidoException {
     Request request = requestBuilder(path)
             .delete()
             .build();
 
     return performRequest(request);
   }
 
-  public FileParam download(String path) throws OnfidoException {
-    throw OnfidoException.networkError(null);
-  }
-
-  public String postUpload(String path, Map<String, Object> params) throws OnfidoException {
-    MultipartBody.Builder builder = new MultipartBody.Builder().setType(MultipartBody.FORM);
+  protected void upload(String path, RequestBody requestBody) throws OnfidoException {
+    Request request = requestBuilder(path)
+            .post(requestBody)
+            .build();
 
-    for (Map.Entry<String, Object> entry : params.entrySet()) {
-      Object value = entry.getValue();
-      if (value instanceof FileParam) {
-        FileParam fileParam = (FileParam) value;
-        builder.addFormDataPart(entry.getKey(), fileParam.getFileName(), createFileBody(fileParam));
-      } else if (value != null) {
-        builder.addFormDataPart(entry.getKey(), value.toString());
-      }
-    }
+    performRequest(request);
+  }
 
+  protected InputStream download(String path) throws IOException {
     Request request = requestBuilder(path)
-      .post(builder.build())
-      .build();
+            .get()
+            .build();
 
-    return performRequest(request);
+    Response response = CLIENT.newCall(request).execute();
+    return response.body().byteStream();
```

Comment:
You'll need to handle api errors and network errors here, like the `performRequest` method is doing (it can't be used directly because it gets the body as a string, but you might be able to tweak it slightly)

Might be worth adding a test too for the case where the api returns an error

Reply:
Done